### ISSUE TITLE
Exception in thread "AWT-EventQueue-0" java.lang.NoClassDefFoundError: ModernDocking/Dockable

### ISSUE THREAD

**Issue by i-make-robots** [andrewauclair/ModernDocking#145]
I am using the api, single app, and ui, all v0.10.0.  
I am running java 21, tho my POM says minimum 15.  
In module-info.java I `requires modern.docking.single.app;`
My test case is the same as Hello World example:

```java
public class RO3 extends JFrame {
    public RO3() {
        setSize(800, 600);
        try {
            Docking.initialize(this);
        } catch(Exception e) {
            e.printStackTrace();
        }

        RootDockingPanel root = new RootDockingPanel(this);
        add(root, BorderLayout.CENTER);

        try {
            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
        } catch(Exception ignored) {}

    }
    public static void main(String[] args) {
        SwingUtilities.invokeLater(() -> (new RO3()).setVisible(true) );
    }
}
```

I get

```
Exception in thread "AWT-EventQueue-0" java.lang.NoClassDefFoundError: ModernDocking/Dockable
	at ModernDocking.api.DockingAPI.<init>(DockingAPI.java:59)
	at modern.docking.single.app@0.10.0/ModernDocking.Docking$1.<init>(Docking.java:46)
	at modern.docking.single.app@0.10.0/ModernDocking.Docking.initialize(Docking.java:46)
	at com.marginallyclever.robotoverlord/com.marginallyclever.ro3.RO3.<init>(RO3.java:16)
	at com.marginallyclever.robotoverlord/com.marginallyclever.ro3.RO3.lambda$main$0(RO3.java:30)
	at java.desktop/java.awt.event.InvocationEvent.dispatch$$$capture(InvocationEvent.java:318)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.lang.ClassNotFoundException: ModernDocking.Dockable
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	... 19 more
```

Please: what am I doing wrong?

**Comment by andrewauclair** [comment]
Do you have api, single-app and ui all in your pom? I think I need to make single-app require api.

**Comment by i-make-robots** [comment]
> Do you have api, single-app and ui all in your pom?
Yes.

> I think I need to make single-app require api.
Wasn't mentioned in the README or the Wiki page for [Hello World](https://github.com/andrewauclair/ModernDocking/wiki/Hello-World).  What do you mean?

**Comment by andrewauclair** [comment]
It's in the README.

**Comment by andrewauclair** [comment]
Can you share the pom? I only use Gradle, but maybe I can notice something.

**Comment by i-make-robots** [comment]
https://github.com/MarginallyClever/Robot-Overlord-App/tree/ro3
`src/main/java/module-info.java`
`src/main/java/com/marginallyclever/ro3/RO3.java`

I see where it says "add api, then add either single or multi".  I see nothing about "make single require api" or what that entails.

**Comment by andrewauclair** [comment]
Sorry, I didn't mean that it works that way right now.

**Comment by andrewauclair** [comment]
> https://github.com/MarginallyClever/Robot-Overlord-App/tree/ro3
> `src/main/java/module-info.java`
> `src/main/java/com/marginallyclever/ro3/RO3.java`

Not seeing it. Let me know when it's up there and I'll look.

**Comment by i-make-robots** [comment]
https://github.com/MarginallyClever/Robot-Overlord-App/blob/ro3/src/main/java/com/marginallyclever/ro3/RO3.java

It tried to be smart and block me because I forgot the commit message. :P  sorry

**Comment by andrewauclair** [comment]
Ok, sorry about that. Turns out I never really made sure modules worked since I don't use them in my project that uses Modern Docking. I have corrected this (hopefully) in the latest 0.11.0-SNAPSHOT build and tested it with your project. I had to move all the classes in modern-docking-single-app to a new package because it conflicted with modern-docking-api and will have to do the same with modern-docking-multi-app.

Below is what I added to your pom.xml and modules-info.java to make this work. Once the real 0.11.0 is out you won't need the snapshot repository part.

pom.xml
```xml
		<repository>
			<id>snapshots-repo</id>
			<url>https://s01.oss.sonatype.org/content/repositories/snapshots</url>
			<releases><enabled>false</enabled></releases>
			<snapshots><enabled>true</enabled></snapshots>
		</repository>

		<dependency>
			<groupId>io.github.andrewauclair</groupId>
			<artifactId>modern-docking-api</artifactId>
			<version>0.11.0-SNAPSHOT</version>
		</dependency>
		<dependency>
			<groupId>io.github.andrewauclair</groupId>
			<artifactId>modern-docking-single-app</artifactId>
			<version>0.11.0-SNAPSHOT</version>
		</dependency>
		<dependency>
			<groupId>io.github.andrewauclair</groupId>
			<artifactId>modern-docking-ui</artifactId>
			<version>0.11.0-SNAPSHOT</version>
		</dependency>
```

and the modules-info.java:

```java
requires modern_docking.single_app;
requires modern_docking.ui_ext;
```

**Comment by i-make-robots** [comment]
Good news, everyone!  It doesn't crash.  I was able to get a Dockable panel working by adding
```java
	requires modern_docking.api;  // new line
	requires modern_docking.single_app;
	requires modern_docking.ui_ext;
```
Thank you for the super fast response!
